### PR TITLE
Ensure that raised AccessError yields 404 response code

### PR DESCRIPTION
**Pull request by acrellin**
Depends on https://github.com/cesium-ml/baselayer/pull/226
Partially addresses https://github.com/skyportal/skyportal/issues/1743
Closes https://github.com/skyportal/skyportal/issues/1380

### CODE REVIEW

File: skyportal/handlers/api/source.py [hunk 1]
```diff
@@ -1489,9 +1489,7 @@ async def get(self, obj_id):
               application/json:
                 schema: Error
         """
-        source = Obj.get_if_accessible_by(obj_id, self.current_user)
-        if source is None:
-            return self.error('Source not found', status=404)
+        source = Obj.get_if_accessible_by(obj_id, self.current_user, raise_if_none=True)
```
Comment (stefanv) [inline skyportal/handlers/api/source.py:1492]: Why is this better than raising the 404 explicitly?

**Review by stefanv — COMMENTED** [review]
Looks OK; I'm not sure I 100% understand why this is necessary.  I left a comment on the baselayer PR which uses `error` instead of adding special logic inside of it.  There, I think you can explicitly examine the Exception type to see if it is an AccessError, and then call `error` appropriately.

File: skyportal/handlers/api/source.py (hunk 1 above)
Comment (acrellin) [reply skyportal/handlers/api/source.py:1492]: checking manually is fine, but we're largely moving over to the more compact pattern here. this is changed to easily test via the new source API test

Comment (stefanv) [reply skyportal/handlers/api/source.py:1492]: OK, automating that makes sense.  Thanks for explaining to me.

**Comment by acrellin** [comment]
@stefanv updated, ready for re-review

**Review by stefanv — APPROVED** [review]

**Comment by acrellin** [comment]
Depends on https://github.com/cesium-ml/baselayer/pull/230

**Comment by acrellin** [comment]
Now I need to update a bunch of tests ;)

**Review by stefanv — APPROVED** [review]

**Comment by acrellin** [comment]
Closing as stalled; will address in a future PR.

**Pull request closed without merging**